extensions/cba_settings/cba_settings.cpp

```diff
@@ -0,0 +1,335 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <algorithm>
+#include "trim.h"
+
+#define VERSION "1.0.0"
+
+#define RET \
+strncpy_s(output, outputSize, result.c_str(), _TRUNCATE);\
+return
+
+#define RETURN(val)\
+strncpy_s(output, outputSize, result.c_str(), _TRUNCATE);\
+return (val)
+
+#define SYNTAX_ERROR_WRONG_PARAMS_SIZE 101
+#define SYNTAX_ERROR_WRONG_PARAMS_TYPE 102
+#define PARAMS_ERROR_TOO_MANY_ARGS 201
+#define EXECUTION_WARNING_TAKES_TOO_LONG 301
+
+#define TEMP_FILE "userconfig/temp"
+#define SETTINGS_FILE "userconfig/cba_settings.sqf"
+#define DELIM ":"
+#define MAX_CALLBACK_LENGTH 10000
+
+using namespace std;
+
+string version() {
+    return VERSION;
+}
+
+string clear() {
+    ofstream file;
+    file.open(SETTINGS_FILE, ios::out);
+    file.close();
+    return "true";
+}
+
+string load(int strlen) {
+    string result = "";
```

Comment:
reserve.

`result.reserve(std::filesystem::size(SETTINGS_FILE));`

Otherwise you resize/reallocate/copy almost every time you append a line.

Reply:
Cool, didn't know `filestream::size` works is a thing. Though, is it reasonable to allocate ~10k characters (bytes?) at once?

Reply:
Well more reasonable than first allocating 16, then deleting and reallocating 32, then deleting and reallocating 64, then deleting and reallocating 128, then deleting and reallocating 512, then deleting and reallocating 2048, then deleting and reallocating 8192, then...